org.eclipse.jdt.core/model/org/eclipse/jdt/internal/compiler/parser/TypeConverter.java

```diff
@@ -443,17 +437,17 @@ private TypeReference decodeType2(char[] typeName, int length, int start, int en
 					   Because of the way type signatures are encoded, TypeConverter.decodeType(String, int, int, int) is immune
 					   to this problem. See https://bugs.eclipse.org/bugs/show_bug.cgi?id=325633
 					 */
-					if (this.has1_5Compliance || includeGenericsAnyway) {
+					if (includeGenericsAnyway) {
```

Comment:
The problem arises from the fact that `(this.has1_5Compliance || includeGenericsAnyway) != (includeGenericsAnyway)` !!!

We are replacing `(true || boolean)` with `(boolean)`